File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20025.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse20025Results
import scala.collection.immutable.Seq

/**
 * 
 * @param results 
 * @param offset 
 * @param number 
 * @param totalResults 
 */
case class InlineResponse20025(results: Seq[InlineResponse20025Results],
                offset: Int,
                number: Int,
                totalResults: Int
                )

object InlineResponse20025 {
    /**
     * Creates the codec for converting InlineResponse20025 from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20025] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20025] = deriveEncoder
}

File: dart/com/spoonacular/client/model/inline_response20025.dart
part of openapi.api;

class InlineResponse20025 {
  
  List<InlineResponse20025Results> results = [];
  
  int offset = null;
  
  int number = null;
  
  int totalResults = null;
  InlineResponse20025();

  @override
  String toString() {
    return 'InlineResponse20025[results=$results, offset=$offset, number=$number, totalResults=$totalResults, ]';
  }

  InlineResponse20025.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['results'] == null) {
      results = null;
    } else {
      results = InlineResponse20025Results.listFromJson(json['results']);
    }
    if (json['offset'] == null) {
      offset = null;
    } else {
          offset = json['offset'];
    }
    if (json['number'] == null) {
      number = null;
    } else {
          number = json['number'];
    }
    if (json['totalResults'] == null) {
      totalResults = null;
    } else {
          totalResults = json['totalResults'];
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (results != null)
      json['results'] = results;
    if (offset != null)
      json['offset'] = offset;
    if (number != null)
      json['number'] = number;
    if (totalResults != null)
      json['totalResults'] = totalResults;
    return json;
  }

  static List<InlineResponse20025> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20025>() : json.map((value) => new InlineResponse20025.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20025> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20025>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20025.fromJson(value));
    }
    return map;
  }
}

